lib/open_trip_planner_client/itinerary_group.ex

```diff
@@ -39,11 +44,53 @@ defmodule OpenTripPlannerClient.ItineraryGroup do
   """
   @spec groups_from_itineraries([Itinerary.t()], Keyword.t()) :: [%__MODULE__{}]
   def groups_from_itineraries(itineraries, opts \\ []) do
+    groups = itineraries |> to_groups(opts)
+
+    opts
+    |> Keyword.get(:ideal_itineraries, [])
+    |> to_groups(opts |> Keyword.put(:available?, false))
```

Comment:
nit: technically we don't know these groups are actually _unavailable_, yet

Reply:
True true. This was kind of a shortcut.

I [added](https://github.com/mbta/open_trip_planner_client/pull/14/commits/3b15384fe93ed939f497fbff6dddac92b60afe25) an explicit `mark_unavailable` step